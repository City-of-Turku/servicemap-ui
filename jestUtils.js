import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { ThemeProvider } from '@mui/material/styles';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import themes from './src/themes';
import finnish from './src/i18n/fi';

const mockStore = configureStore([]);

// Mock props for intl provider
const intlMock = {
  locale: 'fi',
  messages: finnish,
  wrapRichTextChunksInFragment: false,
};

const Providers = (mockState) => ({ children }) => {
  const store = mockStore(mockState);
  return (
    <Provider store={store}>
      <IntlProvider {...intlMock}>
        <ThemeProvider theme={themes.SMTheme}>
          {children}
        </ThemeProvider>
      </IntlProvider>
    </Provider>
  );
}

export const getRenderWithProviders = mockState => component => render(component, { wrapper: Providers(mockState) });