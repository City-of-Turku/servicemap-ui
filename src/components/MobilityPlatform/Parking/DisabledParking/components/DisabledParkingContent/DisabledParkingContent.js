import { Typography } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import useLocaleText from '../../../../../../utils/useLocaleText';

const DisabledParkingContent = ({ classes, intl, item }) => {
  const getLocaleText = useLocaleText();

  return (
    <div className={classes.container}>
      <div className={classes.headerContainer}>
        <Typography variant="subtitle1">{intl.formatMessage({ id: 'mobilityPlatform.content.disabledParking.title' })}</Typography>
      </div>
      <div className={classes.textContainer}>
        <Typography variant="body2">
          {intl.formatMessage({ id: 'mobilityPlatform.content.disabledParking.amount' })}
          :
          {' '}
          {item.extra.invapaikkoja}
        </Typography>
        <Typography variant="body2">
          {intl.formatMessage({ id: 'mobilityPlatform.content.publicParking.amount' })}
          :
          {' '}
          {item.extra.paikkoja_y}
        </Typography>
        <Typography variant="body2">
          {getLocaleText(item.extra.rajoitustyyppi)}
        </Typography>
        {item.extra.saavutettavuus.fi === 'vapaa paasy' ? (
          <Typography variant="body2">
            {intl.formatMessage({ id: 'mobilityPlatform.content.publicParking.access' })}
          </Typography>
        ) : null}
      </div>
    </div>
  );
};

DisabledParkingContent.propTypes = {
  classes: PropTypes.objectOf(PropTypes.any).isRequired,
  intl: PropTypes.objectOf(PropTypes.any).isRequired,
  item: PropTypes.arrayOf(PropTypes.any),
};

DisabledParkingContent.defaultProps = {
  item: [],
};

export default DisabledParkingContent;
