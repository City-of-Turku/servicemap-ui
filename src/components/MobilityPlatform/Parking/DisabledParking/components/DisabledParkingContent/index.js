import { withStyles } from '@material-ui/core';
import { injectIntl } from 'react-intl';
import DisabledParkingContent from './DisabledParkingContent';
import styles from './styles';

export default withStyles(styles)(injectIntl(DisabledParkingContent));
