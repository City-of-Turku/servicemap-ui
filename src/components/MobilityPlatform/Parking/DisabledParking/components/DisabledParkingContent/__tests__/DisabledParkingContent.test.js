/* eslint-disable max-len */
// Link.react.test.js
import React from 'react';
import { getRenderWithProviders } from '../../../../../../../../jestUtils';
import finnishTranslations from '../../../../../../../i18n/fi';
import { initialState } from '../../../../../../../redux/reducers/user';
import DisabledParkingContent from '../index';

const mockProps = {
  item: {
    extra: {
      invapaikkoja: 5,
      paikkoja_y: 5,
      rajoitustyyppi: {
        fi: 'Erityisalue',
      },
      saavutettavuus: {
        fi: 'vapaa paasy',
      },
    },
  },
};

const renderWithProviders = getRenderWithProviders({
  user: initialState,
});

describe('<DisabledParkingContent />', () => {
  it('should work', () => {
    const { container } = renderWithProviders(<DisabledParkingContent {...mockProps} />);
    expect(container).toMatchSnapshot();
  });

  it('does show text correctly', () => {
    const { container } = renderWithProviders(<DisabledParkingContent {...mockProps} />);

    const p = container.querySelectorAll('p');
    const h6 = container.querySelector('h6');
    expect(h6.textContent).toContain(finnishTranslations['mobilityPlatform.content.disabledParking.title']);
    expect(p[0].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.disabledParking.amount']}: ${mockProps.item.extra.invapaikkoja}`);
    expect(p[1].textContent).toContain(`${finnishTranslations['mobilityPlatform.content.publicParking.amount']}: ${mockProps.item.extra.paikkoja_y}`);
    expect(p[2].textContent).toContain(mockProps.item.extra.rajoitustyyppi.fi);
    expect(p[3].textContent).toContain(finnishTranslations['mobilityPlatform.content.publicParking.access']);
  });
});
