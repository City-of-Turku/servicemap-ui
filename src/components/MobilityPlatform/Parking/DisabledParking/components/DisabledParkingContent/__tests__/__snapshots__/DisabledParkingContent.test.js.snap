// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DisabledParkingContent /> should work 1`] = `
<div>
  <div
    class="injectIntl(DisabledParkingContent)-container-1"
  >
    <div
      class="injectIntl(DisabledParkingContent)-headerContainer-2"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1"
      >
        Invapysäköintialue
      </h6>
    </div>
    <div
      class="injectIntl(DisabledParkingContent)-textContainer-3"
    >
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Invapaikkojen määrä
        :
         
        5
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Parkkipaikkoja yhteensä
        :
         
        5
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Erityisalue
      </p>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        Vapaa pääsy
      </p>
    </div>
  </div>
</div>
`;
